pinot-core/src/main/java/org/apache/pinot/core/segment/index/loader/defaultcolumn/V3DefaultColumnHandler.java

```diff
@@ -61,21 +63,27 @@ protected void updateDefaultColumn(String column, DefaultColumnAction action)
           "Default value indices for column: " + column + " cannot be removed for V3 format segment.");
     }
 
-    // Create new dictionary and forward index, and update column metadata
-    createColumnV1Indices(column);
-
-    // Write index to V3 format.
     FieldSpec fieldSpec = _schema.getFieldSpecFor(column);
     Preconditions.checkNotNull(fieldSpec);
     boolean isSingleValue = fieldSpec.isSingleValueField();
-    File dictionaryFile = new File(_indexDir, column + V1Constants.Dict.FILE_EXTENSION);
-    File forwardIndexFile;
-    if (isSingleValue) {
-      forwardIndexFile = new File(_indexDir, column + V1Constants.Indexes.SORTED_SV_FORWARD_INDEX_FILE_EXTENSION);
+
+    Set<String> textIndexColumns = indexLoadingConfig.getTextIndexColumns();
```

Comment:
why do you need to create a raw forward index. This code can be removed

Reply:
Provided there are no assumptions anywhere that (text index == raw forward index).

To start with, it is ok to keep things consistent, and relax the constraints as we move forward.